flink/v1.13/flink/src/main/java/org/apache/iceberg/flink/data/FlinkOrcWriters.java

```diff
@@ -303,6 +303,9 @@ static class RowDataWriter extends GenericOrcWriters.StructWriter<RowData> {
 
     @Override
     protected Object get(RowData struct, int index) {
+      if (index >= struct.getArity()) {
```

Comment:
this assumes schema change only has new fields added in the end. There should be validation during schema refresh and only apply the refresh it satisfy the compatibility rules.